### PR TITLE
Implement transactions by address endpoint

### PR DESCRIPTION
**Pull request by mwadon**

### CODE REVIEW

File: api/model.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export enum SortOrder {
 }
 
 export type Sort<T> = {
-    field: keyof T
+    field: keyof T | string
```
Comment (mwadon) [inline api/model.ts:11]: @kpudlik I want to sort by ordinal which is nested `lastTransactionRef.ordinal`
I set there also a `string` type but maybe you have a better idea?

File: api/elastic.ts [hunk 2]
```diff
@@ -82,13 +101,24 @@ export const getTransactionBySnapshot = (es: Client) => (term: string): TaskEith
     return findAll(getByFieldQuery<Transaction>(ESIndex.Transactions, 'snapshotHash', term, -1)(es))
 }
 
-const findOne = (search: TransportRequestPromise<ApiResponse>) =>
-    pipe(
-        findAll(search),
-        map(s => s[0])
-    )
+export const getTransactionByAddress = (es: Client) => (term: string, field: 'receiver' | 'sender' | null = null): TaskEither<ApplicationError, Transaction[]> => {
```
Comment (marcinwadon) [inline api/elastic.ts:104]: If you want to keep it type safe, you can combine `keyof` and `Pick` there:
```
field: (keyof Pick<Transaction, 'sender' | 'receiver'>) | null
```

---

`Pick<Transaction, 'sender' | 'receiver'>` gives you a type:
```
{ sender: .., receiver: .. }
```
and using `keyof` on this gives you `'sender' | 'receiver'`

File: api/model.ts (hunk 1 above)
Comment (kpudlik) [reply api/model.ts:11]: Typescript doesn't allow that :<

You can technically use something like lenses/path from Ramda: https://ramdajs.com/docs/#path

here you can read more: https://github.com/microsoft/TypeScript/issues/12290

**Review by marcinwadon — APPROVED** [review]

File: api/model.ts (hunk 1 above)
Comment (kpudlik) [reply api/model.ts:11]: In general you could potentially make Sort<Transaction | Transaction["lastTransactionRef"]>

But it wont give you such string:

"lastTransactionRef.ordinal" :p

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**